### PR TITLE
Added checking to make sure that a submitted timestamp is an int.

### PR DESCRIPTION
**Pull request by mayhem**
As it says on the tin. Also included a check to make sure that the timestamp is actually present.

### CODE REVIEW

File: webserver/views/api.py [hunk 1]
```diff
@@ -205,6 +205,13 @@ def get_messybrainz_data(listen):
 
 def _validate_listen(listen):
     """Make sure that required keys are present, filled out and not too large."""
+
+    if not 'listened_at' in listen:
+        _log_raise_400("JSON document must contain the key listened_at at the top level.", listen)
+        
+    if type(listen['listened_at']) != int:
```
Comment (alastair) [inline webserver/views/api.py]: do we want this beahaviour? an alternative would be 
```
try:
    int(listen['listened_at'])
except ValueError:
    raise here
```
which would also let us accept strings from less-robust clients.

Comment (gentlecat) [inline webserver/views/api.py]: I agree. You can just convert it to integer, if that's what we use.

**Comment by alastair** [comment]
awesome, go for it

**Comment by gentlecat** [comment]
:ship: :shipit: :ship:

**Pull request merged**